compiler/ccgexprs.nim

```diff
@@ -2359,6 +2359,7 @@ proc genMove(p: BProc; n: PNode; d: var TLoc) =
     if d.k == locNone: getTemp(p, n.typ, d)
     if p.config.selectedGC in {gcArc, gcAtomicArc, gcOrc}:
       genAssignment(p, d, a, {})
+      resetLoc(p, a)
```

Comment:
Instead of `resetLoc` we need to call `=wasMoved` here.

Reply:
Sure thing

Reply:
Succeeded by https://github.com/nim-lang/Nim/pull/22288